### PR TITLE
Add android led notification

### LINKED ISSUE #128: Android Led notification not working
Hi,

The notifications generated by the plugin does not activate the notification led on Android. Is it possible to add this feature (possibly with the possibility to configure the color of the led) ?

Thx

### PR DESCRIPTION
**Pull request by htbkoo**
Aims to close #128 .

However would require review by @EddyVerbruggen and probably some discussion first :-).

### Caveats
1. Using `setLights` does not work on my OnePlus 5T (with Android 8.1.0) so I followed the [Android documentation](https://developer.android.com/reference/android/app/Notification.Builder.html#setLights(int,%20int,%20int)) to use [`NotificationChannel.enableLights`](https://developer.android.com/reference/android/app/NotificationChannel#enableLights(boolean)) instead as it is suggested in the Android documentation that `setLights` is deprecated since API level 26.
    1. However `NotificationChannel.enableLights` only take 1 boolean parameter (to set whether notifications posted to this channel should display notification lights) so none of the options (`ledColor`, `ledOn`, `ledOff`) are effective
2. The `README.md` and demo have not been updated yet as I believe the might be changes to the propsed API
3. Version is not bumped yet either.

### CODE REVIEW

File: src/local-notifications-common.ts [hunk 1]
```diff
@@ -157,6 +183,16 @@ export interface ScheduleOptions {
...
 
+  /**
+   * Enable the notification LED light with optional LED light style settings
+   * - empty object if you want to use default settings
+   * - or override the settings specifically
+   *
+   * Android only.
+   * Default not set.
+   */
+  notificationLed?: NotificationLedSettings;
```
Comment (htbkoo) [inline src/local-notifications-common.ts]: Not sure if this API is what we want or if we would want to allow the user to just put down a `true` if the user is happy with the default settings (i.e. similar to the `thumbnail` option)
```suggestion
  notificationLed?: boolean | NotificationLedSettings;
```

Of course we would have to update the corresponding usage of this as well :-)

File: native-src/android/app/src/main/java/com/telerik/localnotifications/Builder.java [hunk 2]
```diff
@@ -26,6 +27,10 @@
     private static final String TAG = "Builder";
     private static final String DEFAULT_CHANNEL = "Notifications";
 
+    private static final int DEFAULT_NOTIFICATION_COLOR = Color.parseColor("#ffffffff");
```
Comment (htbkoo) [inline native-src/android/app/src/main/java/com/telerik/localnotifications/Builder.java]: I got these default values from [this StackOverflow answer](https://stackoverflow.com/a/34195759) which I think they got the values from the [Android source code](https://android.googlesource.com/platform/frameworks/base/+/master/core/res/res/values/config.xml#897)

File: native-src/android/app/src/main/java/com/telerik/localnotifications/Builder.java [hunk 3]
```diff
@@ -84,6 +94,17 @@ static Notification build(JSONObject options, Context context, int notificationI
 
     // Notification styles:
 
+    private static void applyNotificationLed(JSONObject options, NotificationCompat.Builder builder) {
+        if (shouldEnableNotificationLed(options)) {
+            try {
+                JSONObject notificationLed = options.getJSONObject("notificationLed");
+                builder.setLights(notificationLed.optInt("ledColor", DEFAULT_NOTIFICATION_COLOR), notificationLed.optInt("ledOn", DEFAULT_NOTIFICATION_LED_ON), notificationLed.optInt("ledOff", DEFAULT_NOTIFICATION_LED_OFF));
```
Comment (htbkoo) [inline native-src/android/app/src/main/java/com/telerik/localnotifications/Builder.java]: Apologies but I am actually quite new - may I ask if this (combined with [this `color=color.android`](https://github.com/EddyVerbruggen/nativescript-local-notifications/pull/133/files#diff-75678843cd805744830e5ddc817a4a44R219)) the correct way to use `Color`?

File: native-src/android/app/src/main/java/com/telerik/localnotifications/Builder.java [hunk 4]
```diff
@@ -84,6 +94,17 @@ static Notification build(JSONObject options, Context context, int notificationI
...
+            } catch (JSONException e) {
+                Log.e(TAG, "Error parsing options.notificationLed", e);
```
Comment (htbkoo) [inline native-src/android/app/src/main/java/com/telerik/localnotifications/Builder.java]: Please feel free to suggest any better way to handle the `JSONException`. :-)

(Currently the only improvement inside my head is to extract the 2 exception handling to a common method)

File: src/local-notifications.android.ts [hunk 5]
```diff
@@ -215,6 +215,10 @@ export class LocalNotificationsImpl extends LocalNotificationsCommon implements
             options.color = options.color.android;
           }
 
+          if (options.notificationLed && options.notificationLed.ledColor){
+            options.notificationLed.ledColor = options.notificationLed.ledColor.android;
```
Comment (htbkoo) [inline src/local-notifications.android.ts]: Please see [this discussion](https://github.com/EddyVerbruggen/nativescript-local-notifications/pull/133/files#r249005541).
I Am trying to follow [this](https://github.com/EddyVerbruggen/nativescript-local-notifications/pull/133/files#diff-75678843cd805744830e5ddc817a4a44R215)

**Comment by EddyVerbruggen** [comment]
Hi! Thanks for looking into this!

Considering the fact that (as you pointed out) a color can no longer be passed with the new level 26 API, I'd say the plugin should only provide a `boolean` to enable the lights. Completely ignore the color in your code.

And in the native (Java) code, check for runtime API level 26 availability at runtime to use either the old or new API. Something similar to what was used [here](https://github.com/EddyVerbruggen/nativescript-local-notifications/blob/9c06edda3369a2000f9b3ad68676059bad9e3a3f/native-src/android/app/src/main/java/com/telerik/localnotifications/ActionGroup.java#L89).

**Comment by jbarnabe** [comment]
Hi @htbkoo, thanks for this.

@EddyVerbruggen, @htbkoo : I take a look at [https://developer.android.com/reference/android/app/NotificationChannel](NotificationChannel)
And it seems possible to configure light color with [https://developer.android.com/reference/android/app/NotificationChannel#setLightColor(int)](NotificationChannel.setLightColor) so I think it's important to keep the setting "ledColor" and call the right methods according to the level of the . No ?

**Comment by EddyVerbruggen** [comment]
@jbarnabe Ah, that's where it went! Yep, then passing in a color property makes sense.

@htbkoo The type can be `Color`, as imported from `tns-core-modules/color`.

**Comment by htbkoo** [comment]
@jbarnabe thanks for the references - yea being able to set the color would be great and I am trying to see how to make it work :)
@EddyVerbruggen Sure I guess this makes more sense :-)

On the other hand I guess we could take away `ledOn` and `ledOff` from the options and only keep `ledColor` (of type `Color`) - I think changing the type to `boolean | Color` (similar to `thumbnail`) may sound more suitable given that now we would only have one option?

**Comment by jbarnabe** [comment]
> On the other hand I guess we could take away ledOn and ledOff from the options and only keep ledColor (of type Color) - I think changing the type to boolean | Color (similar to thumbnail) may sound more suitable given that now we would only have one option?

I agree, I think only one option makes more sense, but it's up to @EddyVerbruggen to confirm.

> Otherwise I am not sure if the color update would be effective as the docs (the one suggested by @jbarnabe) mentioned that it is "only modifiable before the channel is submitted to NotificationManager.createNotificationChannel(NotificationChannel)."

There is the same mention for 'NotificationChannel.enableLights' method. I guess this is the normal behavior and that in production a change is effective only in case of major update of Android (ex: 8.1 to 9.0) or full reinstallation of the application. It's true that it's not great for testing, so you can probably create several NotificationChannel in your test app that includes the new version of the plugin to get around it.

**Comment by EddyVerbruggen** [comment]
One option is fine by me as well guys. Good work!

**Comment by jbarnabe** [comment]
Hi @EddyVerbruggen, any plan for a release with this PR ?

**Comment by EddyVerbruggen** [comment]
@jbarnabe I may have missed that last comment/commit. I'll take care of this real soon!

**Comment by htbkoo** [comment]
@EddyVerbruggen Thanks! 👍 
(Apologies I was quite busy in the last month so I had not followed up to see if anything is pending) :-)

**Comment by EddyVerbruggen** [comment]
Great work, @htbkoo! I've merged it and will publish to npm in a minute.

**Comment by jbarnabe** [comment]
Thanks @htbkoo and @EddyVerbruggen !

**Pull request merged**